src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -53,33 +53,33 @@ export async function computeLpFeeForRefresh(
   gammaCutoff: FlowTupleParameters,
   blockNumber?: number
 ): Promise<BigNumber> {
-  const configStoreClient = hubPoolClient.configStoreClient;
   const erc20 = ERC20__factory.connect(hubPoolTokenAddress, hubPoolClient.hubPool.provider);
-  const [decimals, ethSpokeBalance, hubBalance, hubEquity, spokeTargets] = await Promise.all([
-    erc20.decimals({ blockTag: blockNumber }),
+  // Grab the balances of the spoke pool and hub pool at the given block number.
```

Comment:
This would save us 1 RPC call, but it shouldn't cause that much overhead as the results are being run in parallel. This adds the onus of resolving the decimals on the calling function. Could we instead use a lookup from the  `TOKEN_LOOKUP` class to avoid having to pass decimals into this function?

Reply:
This is a good call. Here's a commit where I load the `decimals` directly from a mapping from the `HubPoolClient`. This results in `decimals` being passed around less from function to function. It also means we can make the `UBABundleState` class simpler I think. 68356887bee75beda8a219929db2fe3e3028e089

Reply:
SGTM

Reply:
This could alternatively have been done via the HubPoolClient itself - it has the complete list of L1 tokens so it's possible to find the decimals and address with an `Array.find` that searches on the token symbol. But TOKEN_SYMBOLS_MAP is also fine.